pkg/sbom/cyclonedx/marshal.go

```diff
@@ -397,15 +431,31 @@ func (e *Marshaler) reportToCdxComponent(r types.Report) (*cdx.Component, error)
 		component.BOMRef = e.newUUID().String()
 	case ftypes.ArtifactFilesystem, ftypes.ArtifactRemoteRepository:
 		component.Type = cdx.ComponentTypeApplication
-		component.BOMRef = e.newUUID().String()
+		// component.BOMRef = e.newUUID().String()
+		// custom logic for consistent report
+		component.BOMRef = bomRef(string(r.ArtifactType), r.ArtifactName)
```

Comment:
Would it make sense to include artifact instance id for uniqueness  similar to how image id is being included in case of image scans?

Reply:
The current artifact name is a combination of application, component, version. I think that should be fine

```
"component": {
      "bom-ref": "filesystem/dfscan:source:3.3.2",
      "type": "application",
      "name": "dfscan:source:3.3.2",
```

Reply:
Cool